src/ResourceManager/Automation/Commands.Automation.Test/ScenarioTests/SourceControlTests.ps1

```diff
@@ -0,0 +1,322 @@
+#
+# Copyright Microsoft Corporation
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+# http://www.apache.org/licenses/LICENSE-2.0
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ----------------------------------------------------------------------------------
+
+# Azure Automation Source Control scenario tests
+# This test suite contains 3 test cases. In each scenario, a source control is created,
+# updated, retrieved and deleted. In addition to this, a sync is started for each source control
+# and then the output streams are retrieved.
+#
+# Here is the sequence:
+# 1) Create source control - validate properties
+# 2) Update the newly created source control - validate that is was updated
+# 3) Start a sync - validate that the sync completes 
+# 4) Retrieve the streams for the completed source control sync job
+# 5) Clean up - remove the source control
+
+# Automation account information
+$resourceGroupName = "frangom-test"
+$subscriptionId = "52d8cf1b-bcac-493a-bbae-f234b5ff38b0"
+$automationAccountName = "frangom-sdkCmdlet-tests"
+
+#region Helper functions
+
+# This function ensure that the given source control does not exist, 
+# which implicitly validates Get-AzureRmAutomationSourceControl and Remove-AzureRmAutomationSourceControl
+function EnsureSourceControlDoesNotExist
+{
+    Param 
+    (
+        [Parameter(Mandatory=$true)]
+        [ValidateNotNullOrEmpty()]
+        [String]
+        $Name
+    )
+
+    $sourceControl = Get-AzureRmAutomationSourceControl -ResourceGroupName $resourceGroupName `
+                                                        -AutomationAccountName $automationAccountName `
+                                                        -Name $Name `
+                                                        -ErrorAction SilentlyContinue
+    if ($sourceControl)
+    {
+        Remove-AzureRmAutomationSourceControl -ResourceGroupName $resourceGroupName `
+                                              -AutomationAccountName $automationAccountName `
+                                              -Name $Name
+    }
+}
+
+function WaitForSourceControlSyncJobState
+{
+    Param 
+    (
+        [Parameter(Mandatory=$true)]
+        [ValidateNotNullOrEmpty()]
+        [String]
+        $Name,
+
+        [Parameter(Mandatory=$true)]
+        [ValidateNotNullOrEmpty()]
+        [Guid]
+        $JobId,
+
+        [Parameter(Mandatory=$true)]
+        [ValidateNotNullOrEmpty()]
+        [String]
+        $ExpectedState
+    )
+
+    $state = ""
+    $waitTimeInSeconds = 5
+    $retries = 20
+
+    do
```

Comment:
There is a common code Retry method that does precisely this

Reply:
Thanks @markcowl, could you please point me to the code?